src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.Workable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private MachineProducer machineProducer;
+
+    @Override
+    public List getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            machineProducer = new BulldozerProducer();
+            return machineProducer.get();
```

Comment:
```suggestion
    public List<Machine> getAll(Class<? extends Machine> type) {
        if (type == Bulldozer.class) {
            return new ArrayList<>(new BulldozerProducer().get());
```

давай не будем использовать не параметризированные List